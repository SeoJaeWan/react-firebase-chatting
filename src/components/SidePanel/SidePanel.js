import React, { Component } from 'react';
import styles from './SidePanel.module.scss';
import Icon from '@material-ui/core/Icon';
import CreateRoomModal from './CreateRoomModal/CreateRoomModal';

class SidePanel extends Component {
  state = {
    CreateRoomModalIsOpen: false,
  };

  openModal = () => {
    this.setState({ CreateRoomModalIsOpen: true });
  };

  closeModal = () => {
    this.setState({ CreateRoomModalIsOpen: false });
  };

  render() {
    const { CreateRoomModalIsOpen } = this.state;
    return (
      <div className={styles['side-panel']}>
        <button onClick={this.openModal} type="button">
          <Icon className={styles.icon}>add_box</Icon>
        </button>
        <CreateRoomModal
          isOpen={CreateRoomModalIsOpen}
          closeModal={this.closeModal}
        />
      </div>
    );
  }
}

export default SidePanel;
