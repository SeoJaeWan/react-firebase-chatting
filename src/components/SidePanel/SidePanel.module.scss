@import "../../styles/variable";

.side-panel {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 2rem 0;
  width: 70px;
  height: 100vh;
  background-color: $color-primary-dark;
}

.icon {
  color: $color-grey-light-2;
  font-size: 48px !important;
  &:hover {
    cursor: pointer;
    opacity: 0.5;
  }
}
