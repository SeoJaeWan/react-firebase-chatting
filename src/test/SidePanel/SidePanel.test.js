import React from 'react';
import SidePanel from '../../components/SidePanel/SidePanel';
import {shallow} from 'enzyme/build';

describe('SidePanel', () => {
  let component = null;
  it('초기 렌더링이 문제없이 되야함', () => {
    component = shallow(<SidePanel/>);
  });

  it('초기 렌더링 스냅샷 일치함', () => {
    expect(component).toMatchSnapshot();
  });
});
