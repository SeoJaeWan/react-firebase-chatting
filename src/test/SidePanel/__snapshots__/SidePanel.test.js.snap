// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidePanel 초기 렌더링 스냅샷 일치함 1`] = `
<div
  className="side-panel"
>
  <button
    onClick={[Function]}
    type="button"
  >
    <WithStyles(Icon)
      className="icon"
    >
      add_box
    </WithStyles(Icon)>
  </button>
  <CreateRoomModal
    closeModal={[Function]}
    isOpen={false}
  />
</div>
`;
